### DISCUSSION TITLE
Slash Being Removed From ChannelMap

### DISCUSSION THREAD

**Discussion by nickboxx** [nextgenhealthcare/connect#5737] [Q&A] [answered]
I have a variable set up with a "\" in it but when you channelMap the variable it removes the "\" like it is never there. I am trying to use this variable as the delimiter is "\.br\" in something I am trying to parse through but I cannot get it to recognize the slash.

Also, when I include the 2nd slash after the "br" it doesn't even acknowledge it as a string anymore. I would assume this is likely because it thinks of the slash in the sense of \n or \r or something like that?

    var lineBreakStr = "\.br"
    channelMap.put('lineBreakStr', lineBreakStr)

**Comment by jonbartels** [comment, marked as answer]
```
var lineBreakStr = "\\.br"
channelMap.put('lineBreakStr', lineBreakStr)
```

**Reply by mjtrac** [reply]
To once again add to Jon's correct comment, the backslash is used in various programming languages to mean "this next character doesn't get its usual meaning, instead, backslash-something means "use this combo as something special."  Backslash-backslash means, "hey, don't treat the second backslash in its usual way, as altering the meaning of the next character.  Instead, treat that first backslash as meaning the second backslash should just be treated as a plain old backslash."  

I've seen the need for four backslashes in at times to get one out.  I've never seen a need for eight, though. :)
